tests/test_core.py

```diff
@@ -2519,6 +2519,7 @@ def test_pthread_proxying(self):
 
   @node_pthreads
   def test_pthread_dispatch_after_exit(self):
+    self.set_setting('EXIT_RUNTIME')
```

Comment:
was this an oversight before? should we test both modes?

Reply:
Without `EXIT_RUNTIME` this test doesn't actually exit correctly (at least not after this change).

In general, any test that uses pthreads on node currently as to set `EXIT_RUNTIME` (see all the other @node_pthreads tests in this file).

The difference was that `emscripten_force_exit()` used to work even without `EXIT_RUNTIME` ... but with this change `maybeExit` basically doesn't do anything unless you set `EXIT_RUNTIME` .. which is think is correct.

Reply:
> Without EXIT_RUNTIME this test doesn't actually exit correctly (at least not after this change).

I have not fully understood the change then, I thought it did not change observable behavior. What specifically might break for users?

Reply:
This is change to semantic only for `node + pthreads + EXIT_RUNTIME=0`.   Normally such programs never exit, or at least never bring down the entire not process, which kind of makes sense since `EXIT_RUNTIME=0`.

However, prior to this change there was an exception for `emscripten_force_exit()` there it did (inadvertently) work.  This change makes `EXIT_RUNTIME=0` take presence of over `emscripten_force_exit()`.   `emscripten_force_exit()` kind of doesn't make sense with `EXIT_RUNTIME=0` anyway, since the code for performing that actual exit is not compiled in.

Also, note that this is really only node we are talking about here.    Which means most users are the emscripten test suite (I think).

Reply:
I see, thanks. I'm not that worried about that combination, in that case.

However, why is this only noticeable in pthreads mode? I don't see pthreads in the PR title or description so it's a surprising thing to me.

Side note, if these modes are complex, I think we could get away with pthreads forcing `EXIT_RUNTIME=1`. The code size hit is small there relative to pthreads.

Reply:
Yes we might want to consider forcing `EXIT_RUNTIME=1` as the default for pthreads builds.. on the other hand is only really node that needs this.. so we might not want to open web users into `EXIT_RUNTIME=1`